pika/adapters/select_connection.py

```diff
@@ -27,20 +27,25 @@
 ERROR = 0x0008
 
 
-if pika.compat.PY2:
-    _SELECT_ERROR = select.error
-else:
-    # select.error was deprecated and replaced by OSError in python 3.3
-    _SELECT_ERROR = OSError
 
+_SELECT_ERROR_CHECKERS = {
+    select.error : lambda e: e.args[0] == errno.EINTR,
+    OSError      : lambda e: e.errno == errno.EINTR,
+    IOError      : lambda e: e.errno == errno.EINTR,
+    }
 
-def _get_select_errno(error):
-    if pika.compat.PY2:
-        assert isinstance(error, select.error), repr(error)
-        return error.args[0]
-    else:
-        assert isinstance(error, OSError), repr(error)
-        return error.errno
+if pika.compat.PY3:
+    # InterruptedError is undefined in PY2
+    #pylint: disable=E0602
+    _SELECT_ERROR_CHECKERS[InterruptedError] = lambda e: True
+
+# We can reduce the number of elements 
```

Comment:
This discussion lead to me notice a bug in _is_resumable() implementation. It is possible that an exception which is a subclass of a classe in _SELECT_ERRORS but whose actual class is not listed in _SELECT_ERROR_CHECKERS. That exception needs to be reraised rather than causing KeyError when _SELECT_ERROR_CHECKERS is indexed.